tests/h/services/search_index/service_test.py

```diff
@@ -77,25 +147,59 @@ def AnnotationSearchIndexPresenter(self, patch):
 
 
 class TestDeleteAnnotationById:
-    @pytest.mark.parametrize("target_index", (None, "another"))
     @pytest.mark.parametrize("refresh", (True, False))
-    def test_delete_annotation(self, search_index, es_client, target_index, refresh):
-        search_index.delete_annotation_by_id(
-            sentinel.annotation_id, target_index, refresh
-        )
+    def test_delete_annotation(self, search_index, es_client, refresh):
+        search_index.delete_annotation_by_id(sentinel.annotation_id, refresh)
 
         es_client.conn.index.assert_called_once_with(
-            index=es_client.index if target_index is None else target_index,
+            index=es_client.index,
             doc_type=es_client.mapping_type,
             body={"deleted": True},
             id=sentinel.annotation_id,
             refresh=refresh,
         )
 
+    @pytest.mark.usefixtures("with_reindex_in_progress")
+    @pytest.mark.parametrize("refresh", (True, False))
+    def test_it_calls_elasticsearch_again_for_a_reindex(
+        self, search_index, es_client, refresh
+    ):
+        search_index.delete_annotation_by_id(sentinel.annotation_id, refresh)
+
+        assert es_client.conn.index.call_count == 2
+        es_client.conn.index.assert_called_with(
+            index="another_index",
+            doc_type=es_client.mapping_type,
+            body=Any(),
+            id=sentinel.annotation_id,
+            refresh=refresh,
+        )
```

Comment:
It might be nice to remove the duplication here? Or is that worse than having it? Suggestions welcome

Reply:
So `es_client.conn.index()` is actually expected to be called twice? But we only care about one of the calls?

You can do something like:

```python
assert es_client.conn.index.call_args_list == [
    h_matchers.Any(),
    mock.call(index="another_index", ...),
]
```

Reply:
Yes, but we also technically don't care about the order. If we did the other one first, that would be fine. Interestingly we can actually cater for that with `h-matchers` too with:

```
assert es_client.conn.index.call_args_list == Any.list.containing([
    h_matchers.Any(),
    mock.call(index="another_index", ...),
]).only()
```

The matching is smart enough to swap the elements if it needs to to get a match. Worth doing?

Reply:
Hmm, I'm not going to do this. It's actually longer (code)than doing it the dumb way, and much harder to explain. it requires you to know more about `h_matchers` to understand.

Once formatted it ends up like this:
```
        assert (
            es_client.conn.index.call_args_list
            == Any.list.containing(
                [
                    Any(),
                    call(
                        index="another_index",
                        doc_type=es_client.mapping_type,
                        body=Any(),
                        id=annotation.id,
                        refresh=False,
                    ),
                ]
            ).only()
        )
```

I don't think that's clearer.